### PR TITLE
SKIP UPGRADE TEST init commit

### PR DESCRIPTION
**Pull request by Khuzaima05**
### Description

Migrating iam-serviceid-apikey-secrets-manager-module to public

issue: https://github.ibm.com/GoldenEye/issues/issues/6299

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- When merging, use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author. The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).
- Merge by using "Squash and merge".

### CODE REVIEW

File: .github/settings.yml [hunk 1]
```diff
@@ -15,14 +15,14 @@ repository:
   # By changing this field, you rename the repository.
 
   # Uncomment this name property and set the name to the current repo name.
-  # name: ""
+  name: "terraform-ibm-iam-serviceid-apikey-secrets-manager-module"
```
Comment (ocofaigh) [inline .github/settings.yml:18]: drop the `-module` from the repo name

File: README.md [hunk 2]
```diff
@@ -1,110 +1,108 @@
...
-Update status and "latest release" badges:
-  1. For the status options, see https://github.ibm.com/GoldenEye/documentation/blob/master/status.md
-  2. Update the "latest release" badge to point to the correct module's repo. Replace "module-template" in two places.
--->
-[![Incubating (Not yet consumable)](https://img.shields.io/badge/status-Incubating%20(Not%20yet%20consumable)-red)](https://terraform-ibm-modules.github.io/documentation/#/badge-status)
-[![latest release](https://img.shields.io/github/v/release/terraform-ibm-modules/terraform-ibm-module-template?logo=GitHub&sort=semver)](https://github.com/terraform-ibm-modules/terraform-ibm-module-template/releases/latest)
+## IAM Service ID API key Secrets Manager module
+
+[![Graduated (Supported)](https://img.shields.io/badge/Status-Graduated%20(Supported)-brightgreen)](https://terraform-ibm-modules.github.io/documentation/#/badge-status)
+[![latest release](https://img.shields.io/github/v/release/terraform-ibm-modules/terraform-ibm-iam-serviceid-apikey-secrets-manager-module?logo=GitHub&sort=semver)](https://github.com/terraform-ibm-modules/terraform-ibm-iam-serviceid-apikey-secrets-manager-module/releases/latest)
```
Comment (ocofaigh) [inline README.md:4]: `-module` has been removed from repo name so need to update in 2 places here

File: README.md [hunk 3]
```diff
@@ -1,110 +1,108 @@
...
 
+In addition to rotation when the TTL or lease expires, you can create a rotation policy to trigger rotation before the leasing time expires. Triggering rotation before expiration provides transition time between the older and the new credentials. You enable this setting by setting `sm_iam_secret_auto_rotation` to `true` and configuring the related rotation variables `sm_iam_secret_auto_rotation_unit` and `sm_iam_secret_auto_rotation_interval`.
 
-<!-- Below content is automatically populated via pre-commit hook -->
 <!-- BEGIN OVERVIEW HOOK -->
 ## Overview
-* [terraform-ibm-module-template](#terraform-ibm-module-template)
+* [terraform-ibm-iam-serviceid-apikey-secrets-manager-module](#terraform-ibm-iam-serviceid-apikey-secrets-manager-module)
```
Comment (ocofaigh) [inline README.md:17]: re-run the hook as the repo has been renamed

File: README.md [hunk 4]
```diff
@@ -1,110 +1,108 @@
...
--->
-
 ```hcl
-
+provider "ibm" {
+  ibmcloud_api_key     = "XXXXXXXXXXXXXX"
+  region               = "us-south"
+}
+module "dynamic_serviceid_apikey1" {
+  source                               = "terraform-ibm-modules/terraform-ibm-iam-serviceid-apikey-secrets-manager-module/ibm"
```
Comment (ocofaigh) [inline README.md:34]: when you onboard to hashicorp, it strips the `terraform-ibm` from the name, so this should be: `"terraform-ibm-modules/iam-serviceid-apikey-secrets-manager/ibm"`

File: README.md [hunk 5]
```diff
@@ -1,110 +1,108 @@
...
+  version                              = "X.X.X"  # Replace "X.X.X" with a release version to lock into a specific release
+  resource_group_id                    = "xxXXxxXXxXxXXXXxxXxxxXXXXxXXXXX"
+  region                               = "us-south"
+  sm_iam_secret_name                   = var.sm_iam_secret_name
```
Comment (ocofaigh) [inline README.md:38]: Don't use `var.` in usage - use a real like value. Same for below variables

File: README.md [hunk 6]
```diff
@@ -1,110 +1,108 @@
...
-# Terraform Modules Template Project
-
-<!--
-Update status and "latest release" badges:
-  1. For the status options, see https://github.ibm.com/GoldenEye/documentation/blob/master/status.md
-  2. Update the "latest release" badge to point to the correct module's repo. Replace "module-template" in two places.
--->
-[![Incubating (Not yet consumable)](https://img.shields.io/badge/status-Incubating%20(Not%20yet%20consumable)-red)](https://terraform-ibm-modules.github.io/documentation/#/badge-status)
-[![latest release](https://img.shields.io/github/v/release/terraform-ibm-modules/terraform-ibm-module-template?logo=GitHub&sort=semver)](https://github.com/terraform-ibm-modules/terraform-ibm-module-template/releases/latest)
+## IAM Service ID API key Secrets Manager module
```
Comment (ocofaigh) [inline README.md:1]: remove one of these `#` symbols

File: examples/complete-no-rotation-policy/README.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
+## Complete with no rotation example
```
Comment (ocofaigh) [inline examples/complete-no-rotation-policy/README.md:1]: remove one of these `#` symbols

File: README.md (hunk 3 above)
Comment (Khuzaima05) [reply README.md:17]: @ocofaigh , I have change the repo name in .github/settings.yml : `terraform-ibm-iam-serviceid-apikey-secrets-manager-module` -> `terraform-ibm-iam-serviceid-apikey-secrets-manager` , But its not populating overview hook with new name.

Comment (Khuzaima05) [reply README.md]: @ocofaigh , here after modification receiving this error:



```
Add module repository to go.mod..........................................Failed
- hook id: go-mod-module-repository
- files were modified by this hook


current value: module github.com/terraform-ibm-modules/terraform-ibm-iam-serviceid-apikey-secrets-manager-module
new value    : module github.com/terraform-ibm-modules/terraform-ibm-iam-serviceid-apikey-secrets-manager

warning: If repository name has changed, then update 'remote.origin.url' locally by running 'git remote set-url origin new_repo_url' or re-clone the repo using the new repo name.
```
But the repo url is github.com/terraform-ibm-modules/terraform-ibm-iam-serviceid-apikey-secrets-manager-module

**Comment by ocofaigh** [comment]
@Khuzaima05 I have renamed the repo to `terraform-ibm-iam-serviceid-apikey-secrets-manager` can you re-clone and address failing pre-commit hooks please?

**Comment by Khuzaima05** [comment]
> @Khuzaima05 I have renamed the repo to `terraform-ibm-iam-serviceid-apikey-secrets-manager` can you re-clone and address failing pre-commit hooks please?

@ocofaigh , I recloned the repo using name `terraform-ibm-iam-serviceid-apikey-secrets-manager` and pre-commit is passing in local
Screenshot 2023-12-05 at 10 29 45 PM

But in pipeline its failing 

Screenshot 2023-12-05 at 10 30 04 PM

**Comment by ocofaigh** [comment]
Oh, something is renaming the repo back to `terraform-ibm-iam-serviceid-apikey-secrets-manager-module` - @Khuzaima05 can you create a separate PR just to update the repo name in `.github/settings.yml` ?

**Comment by Khuzaima05** [comment]
> Oh, something is renaming the repo back to `terraform-ibm-iam-serviceid-apikey-secrets-manager-module` - @Khuzaima05 can you create a separate PR just to update the repo name in `.github/settings.yml` ?

Here is separate PR : https://github.com/terraform-ibm-modules/terraform-ibm-iam-serviceid-apikey-secrets-manager-module/pull/6

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**